libxrdp/xrdp_caps.c

```diff
@@ -360,6 +360,33 @@ xrdp_caps_process_brushcache(struct xrdp_rdp *self, struct stream *s,
     return 0;
 }
 
+/*****************************************************************************/
+static int
+xrdp_caps_process_glyphcache(struct xrdp_rdp *self, struct stream *s,
+                             int len)
+{
+    int glyph_support_level;
+
+    if (len < 40 + 4 + 2 + 2)
```

Comment:
I think this comes from GlyphCache (40bytes) + FlagCache (4bytes) + GlyphSupportLevel (2bytes) + padding (2bytes).  Magic numbers are OK here but please leave a reference "MS-RDPBCGR 2.2.7.1.8" in comment. Like done in `xrdp_constants.h`.